bot2-core/cmake/lcmtypes.cmake

```diff
@@ -466,20 +449,29 @@ function(lcmtypes_install_types)
 endfunction()
 
 macro(lcmtypes_build)
-    find_package(PkgConfig REQUIRED)
-    pkg_check_modules(LCM REQUIRED lcm)
-    
-    #find lcm-gen (it may be in the install path)
-    find_program(LCM_GEN_EXECUTABLE lcm-gen ${EXECUTABLE_OUTPUT_PATH} ${EXECUTABLE_INSTALL_PATH})
-    if (NOT LCM_GEN_EXECUTABLE)
-    	message(FATAL_ERROR "lcm-gen not found!\n")
-    	return()
+    find_package(lcm REQUIRED)
+    if(TARGET lcm::lcm-gen)
+      set(lcm_namespace "lcm::")
```

Comment:
If a recent version of LCM is being used, it will set `LCM_NAMESPACE`. Use that everywhere you are using `lcm::` and by default `LCM_NAMESPACE` is set to `""` so the TARGET is called simply `lcm-gen`.